Assets/Scripts/Door.cs

```diff
@@ -62,22 +90,18 @@ private void Awake()
     void Start()
     {
         animator = GetComponent<Animator>();
-        animator.SetInteger("Type", doorType);
+        animator.SetInteger(Animator.StringToHash(animTypeBool),
+            (int)doorType);
+        audioSource = GetComponent<AudioSource>();
     }
 
-    /// <summary>
-    /// Add to the current solution
-    /// </summary>
-    /// <param name="entry">The solution piece connected to the activated switch</param>
+    // Functionality matches that described in IPuzzle
     public void AppendSolution(char entry)
```

Comment:
Let's just use the property for the CurrentSolution (allow it to be publicly set)